### ISSUE TITLE
Division of `int` does not match division of `np.int64`

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#16453]
### Reproducing code example:

```python
>>> import numpy as np
>>> big_int = 2**53 + 1
>>> big_int64 = np.int64(big_int)
>>> big_int / (big_int * 1000)
0.001
>>> big_int64 / (big_int64 * 1000)
0.0009999999999999998
```

The cause here is that numpy computes `float(x) / float(y)`, but python does something cleverer.

I don't know if we want to fix this, but if we don't we certainly need to document it.

**Comment by ewmoore** [comment]
There is a decent looking description of the algorithm used in the CPython source 

https://github.com/python/cpython/blob/master/Objects/longobject.c#L3730

**Comment by eric-wieser** [comment]
Indeed, it was that source that made me suspicious of our implementation

**Comment by seberg** [comment]
I guess step 3 would be the relevant part there? For Python its a bit more complicated due to the arbitrary precision nature. I wonder if, otherwise, we would have a nice place to document this... I guess it could go into the notes of divide.
Otherwise, I think it would also be OK to add a specific integer true-divide loop without relying on casting to double.
